gnssanalysis/gn_download.py

```diff
@@ -445,6 +479,59 @@ def download_most_recent(
             return ret_vars
 
 
+def download_file_from_cddis(
+    filename: str,
+    ftp_folder: str,
+    output_folder: _Path,
+    ftps: _Optional[_FTP_TLS] = None,
+    max_retries: int = 3,
+    uncomp: bool = True,
+) -> None:
+    """Downloads a single file from the cddis ftp server.
+
+    :param filename: Name of the file to download
+    :ftp_folder: Folder where the file is stored on the remote
+    :output_folder: Folder to store the output file
+    :ftps: Optional active connection object which is reused
+    :max_retries: Number of retries before raising error
+    :uncomp: If true, uncompress files on download
+    """
+    with ftp_tls_cddis(ftps) as ftps:
+        ftps.cwd(ftp_folder)
+        retries = 0
+        download_done = False
+        while not download_done and retries <= max_retries:
+            try:
+                logging.info(f"Attempting Download of: {filename}")
+                check_n_download(filename, str(output_folder) + "/", ftps, uncomp=uncomp)
+                download_done = True
+                logging.info(f"Downloaded {filename}")
+            except _ftplib.all_errors as e:
+                retries += 1
+                if retries > max_retries:
+                    logging.warning(f"Failed to download {filename} and reached maximum retry count ({max_retries}).")
+                    if (output_folder / filename).is_file():
+                        (output_folder / filename).unlink()
+                    raise e
+
+                logging.debug(f"Received an error ({e}) while try to download {filename}, retrying({retries}).")
+                # Add some backoff time (exponential random as it appears to be contention based?)
+                _time.sleep(_random.uniform(0.0, 2.0**retries))
+
+
+def download_multiple_files_from_cddis(files: [str], ftp_folder: str, output_folder: _Path) -> None:
+    """Downloads multiple files in a single folder from cddis in a thread pool.
+
+    :param files: List of str filenames
+    :ftp_folder: Folder where the file is stored on the remote
+    :output_folder: Folder to store the output files
+    """
+    with _concurrent.futures.ThreadPoolExecutor() as executor:
+        # Wrap this in a list to force iteration of results and so get the first exception if any were raised
+        list(executor.map(download_file_from_cddis, files, _repeat(ftp_folder), _repeat(output_folder)))
+
+
+# TODO: Deprecate? Only supports legacy filenames
```

Comment:
Yeah, I think once we have the other functions that allow `download_file_from_cddis` to carry out the expected file downloads, we can definitely get rid of this.

Other functions here I mean stuff like filename generation